Automattic/wp-calypso
# Payments: Convert BillingHistoryList to TypeScript

## Proposed Changes

This PR cleans up the `BillingHistoryList` component and its dependencies, and converts them all to TypeScript.

This was not a straightforward conversion because the component and its functions used a number of techniques that were highly dynamic. In order to complete the conversion, I had to simplify many of these operations.

<img width="1060" alt="Screenshot 2023-03-27 at 3 15 16 PM" src="https://user-images.githubusercontent.com/2036909/228043915-c21cbf33-76aa-45f7-8176-e05e20605bc0.png">

## Testing Instructions

Use an account which has a purchase history.

View `/me/purchases/billing` with and without this PR and make sure that they show the same results.

Select some of the filters (Month and App) at the top of the page and verify that they work as expected.
